Lib/collections/__init__.py

```diff
@@ -360,6 +363,14 @@ def namedtuple(typename, field_names, *, rename=False, module=None):
             raise ValueError(f'Encountered duplicate field name: {name!r}')
         seen.add(name)
 
+    field_defaults = {}
+    if defaults is not None:
+        defaults = tuple(defaults)
+        if len(defaults) > len(field_names):
+            raise TypeError('Got more default values than field names')
+        field_defaults = dict(reversed(list(zip(reversed(field_names),
```

Comment:
Wouldn't the following code be more clear?
```python
if defaults:
    field_defaults = dict(zip(field_names[-len(defaults)], defaults))
```

Reply:
I don't like that the zero-length case has to be guarded against.  It may be a matter of taste, but I prefer the current code over negative indexing.

Reply:
But your code already contains special guards for None. You could just change the one of guards.